Libraries/LibCore/EventLoop.cpp

```diff
@@ -519,6 +519,18 @@ void EventLoop::unregister_signal(int handler_id)
         s_signal_handlers.remove(remove_signo);
 }
 
+void EventLoop::tear_down()
+{
+    s_main_event_loop = nullptr;
+    s_event_loop_stack->clear();
+    s_timers->clear();
+    s_notifiers->clear();
+    s_signal_handlers.clear();
+    s_handling_signal = 0;
+    s_next_signal_id = 0;
+    s_pid = 0;
+}
+
```

Comment:
I'm not so sure about this; it feels like a mess.
The main reason why I'm doing this is that signals are not delivered to forks, and I would like them to be delivered to forks (in specific conditions of course).
 
cc @tomuta (`s_pid` and signals and stuff), @awesomekling (this _is_ a `Core::EventLoop` change)

Any (better) alternative solutions?

Reply:
So the `s_pid` is basically only used to avoid accidently redirecting a signal sent to a fork to the parent. That's because the signal notification uses a pipe, which is inherited by the fork. Checking the current PID against `s_pid` when a signal is delivered allows it to detect whether it was forked and therefore can't send it through the existing pipe (the other end of that pipe is still read by the parent).

Reply:
If you expect the event loop to work in a fork you'd have to close the EventLoop's notification pipe and recreate it in the fork. Otherwise the event loop in the fork also going to notify the parent process, which is quite weird and probably going to result in odd behavior. For example, boost has a notify_fork method that is supposed to be called before and after fork, but due to the odd nature of fork it can deadlock (if multi-threaded). But its purpose is to recreate these types of pipes. But even boost isn't getting this right, because there is a window where a signal can be sent to the fork before notify_fork was called, and then the signal notification will unexpectedly leak to the parent as well. Hence the check against `s_pid` in the signal handler, which closes that loophole.

Reply:
Well, I want to have a fresh event loop in the fork, no notifiers, no deferred invocations from the parent, etc.
I'm not against recreating the signal handlers.

So...so long as I recreate the signal pipe, everything signal-handler related is honky dory?

Reply:
Well, if you're not using threads then you might be able to re-create a new event loop. If you're using threads the nature of `fork` kind of dictates an `exec` (this is true even on Linux).
Last time I looked at the code (that was to add the `s_pid` fix) the `EventLoop` wasn't really designed with `fork` in mind. We'd have to add functionality similar to boost's notify_fork which, among other things, would recreate the notification pipe in a fork. But the `s_pid` check is still necessary due to the window between `fork` and `notify_fork`.

Reply:
No threads here, so I guess we can call this okay and deal with `notify_fork()` and co when we need it...probably.

Reply:
If you only want to get the signal delivery working, you could create a new notification pipe when you detect the `s_pid` mismatch and use that from that point on. But anything else might be wonky, such as pending asynchronous stuff that might be carried into the fork. Which is why boost has the `notify_fork` API, which does the appropriate changes internally.

Reply:
I decided to add a barebones `notify_forked()` that just tears down the event loop (renews pipes, clears timers, etc) in the child.
we can flesh it out or change it whenever we need the extra functionality.